python/pyspark/mllib/linalg/__init__.py

```diff
@@ -764,10 +764,12 @@ def __getitem__(self, index):
         if not isinstance(index, int):
             raise TypeError(
                 "Indices must be of type integer, got type %s" % type(index))
+
+        original_index = index
         if index < 0:
             index += self.size
         if index >= self.size or index < 0:
-            raise ValueError("Index %d out of bounds." % index)
+            raise ValueError("Index %d out of bounds." % original_index)
```

Comment:
It might be better to move this check to line 768, before index gets modified.  That way, you don't have to even create a copy of index (though you'll need to adjust the check's index range).